From 7351db2e9d242a396ffc8c2365a18451fb51e022 Mon Sep 17 00:00:00 2001
Message-Id: <7351db2e9d242a396ffc8c2365a18451fb51e022.1421709247.git.chang-joon.lee@intel.com>
In-Reply-To: <3650eb44734ad4fbe098313349da8185595a20d8.1421709247.git.chang-joon.lee@intel.com>
References: <3650eb44734ad4fbe098313349da8185595a20d8.1421709247.git.chang-joon.lee@intel.com>
From: Shobhit Kumar <shobhit.kumar@intel.com>
Date: Wed, 3 Sep 2014 22:34:53 +0530
Subject: [PATCH 08/75] MUST_REBASE [VPG]: drm/i915/adf: Add wrapper function
 to export DSI specififc data from VBT

Parsed MIPI data from VBT is exposed to the intel adf driver

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need to
work on i915 and ADF convergence path before it can be upstreamed

For: GMINL-3547
Change-Id: I664f0ff5b4bee0926ed24f2d729239be81ad17b1
Signed-off-by: Shobhit Kumar <shobhit.kumar@intel.com>
---
 drivers/gpu/drm/i915/i915_adf_wrapper.c |   11 +++++++++++
 include/drm/i915_adf.h                  |    2 ++
 2 files changed, 13 insertions(+)

diff --git a/drivers/gpu/drm/i915/i915_adf_wrapper.c b/drivers/gpu/drm/i915/i915_adf_wrapper.c
index 195ded7..10b1646 100644
--- a/drivers/gpu/drm/i915/i915_adf_wrapper.c
+++ b/drivers/gpu/drm/i915/i915_adf_wrapper.c
@@ -112,4 +112,15 @@ void intel_adf_dpio_sideband_rw(u32 operation, u32 port, u32 reg, u32 *val)
 }
 EXPORT_SYMBOL(intel_adf_dpio_sideband_rw);
 
+void intel_adf_get_dsi_vbt_data(void **vbt_data, struct drm_display_mode **mode)
+{
+	if (!i915_adf_dev)
+		return;
+
+	*vbt_data = (void *) &i915_adf_dev->vbt.dsi;
+	*mode = i915_adf_dev->vbt.lfp_lvds_vbt_mode;
+}
+EXPORT_SYMBOL(intel_adf_get_dsi_vbt_data);
+
+
 #endif
diff --git a/include/drm/i915_adf.h b/include/drm/i915_adf.h
index 02c5aec..2afec81 100644
--- a/include/drm/i915_adf.h
+++ b/include/drm/i915_adf.h
@@ -41,6 +41,8 @@ extern void intel_adf_dpio_sideband_rw(u32 operation, u32 port,
 extern void intel_adf_pci_sideband_rw(u32 operation, u32 port,
 				      u32 reg, u32 *val);
 extern struct pci_dev *i915_adf_get_pci_dev(void);
+extern void intel_adf_get_dsi_vbt_data(void **vbt_data,
+				   struct drm_display_mode **mode);
 
 #endif
 
-- 
1.7.9.5

